common/rpc/encryption/localStoreTlsProvider.go

```diff
@@ -225,31 +229,33 @@ func (s *localStoreTlsProvider) getOrCreateConfig(
 func newServerTLSConfig(
 	certProvider CertProvider,
 	perHostCertProviderMap PerHostCertProviderMap,
+	config *config.GroupTLS,
 ) (*tls.Config, error) {
 
-	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider)
+	clientAuthRequired := config.Server.RequireClientAuth
+	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
 	if err != nil {
 		return nil, err
 	}
 
 	tlsConfig.GetConfigForClient = func(c *tls.ClientHelloInfo) (*tls.Config, error) {
 		if perHostCertProviderMap != nil {
-			perHostCertProvider, err := perHostCertProviderMap.GetCertProvider(c.ServerName)
+			perHostCertProvider, clientAuth, err := perHostCertProviderMap.GetCertProvider(c.ServerName)
```

Comment:
[Nit] having a variable called clientAuth and a variable called clientAuthRequired makes this a bit hard to decipher. Even weirder is the decision to use one or the other is contigent on perHostCertProvider being nil or not.